Core/Src/porting/common.c

```diff
@@ -67,3 +69,206 @@ void odroid_audio_mute(bool mute)
+    rg_app_desc_t *app = odroid_system_get_app();
+    static emu_speedup_t last_speedup = SPEEDUP_1_5x;
+    static int8_t last_key = -1;
+    static bool pause_pressed = false;
+    static bool macro_activated = false;
+
+    if(joystick->values[ODROID_INPUT_VOLUME]){  // PAUSE/SET button
+        // PAUSE/SET has been pressed, checking additional inputs for macros
+        pause_pressed = true;
+        if(last_key < 0) {
+            if (joystick->values[ODROID_INPUT_POWER]){
+                // Do NOT save-state and then poweroff
+                last_key = ODROID_INPUT_POWER;
+                HAL_SAI_DMAStop(&hsai_BlockA1);
+                odroid_system_sleep();
+            }
+            else if(joystick->values[ODROID_INPUT_START]){ // GAME button
+                // Reserved for future use
+                last_key = ODROID_INPUT_START;
+            }
+            else if(joystick->values[ODROID_INPUT_SELECT]){ // TIME button
+                // Toggle Speedup
+                last_key = ODROID_INPUT_SELECT;
+                if(app->speedupEnabled == SPEEDUP_1x) {
+                    app->speedupEnabled = last_speedup;
+                }
+                else {
+                    last_speedup = app->speedupEnabled;
+                    app->speedupEnabled = SPEEDUP_1x;
+                }
+            }
+            else if(joystick->values[ODROID_INPUT_LEFT]){
+                // Volume Up
+                last_key = ODROID_INPUT_LEFT;
+                int8_t level = odroid_audio_volume_get();
+                if (level > ODROID_AUDIO_VOLUME_MIN) odroid_audio_volume_set(--level);
+            }
+            else if(joystick->values[ODROID_INPUT_RIGHT]){
+                // Volume Down
+                last_key = ODROID_INPUT_RIGHT;
+                int8_t level = odroid_audio_volume_get();
+                if (level < ODROID_AUDIO_VOLUME_MAX) odroid_audio_volume_set(++level);
+            }
+            else if(joystick->values[ODROID_INPUT_UP]){
+                // Brightness Up
+                last_key = ODROID_INPUT_UP;
+                int8_t level = odroid_display_get_backlight();
+                if (level < ODROID_BACKLIGHT_LEVEL_COUNT - 1) odroid_display_set_backlight(++level);
+            }
+            else if(joystick->values[ODROID_INPUT_DOWN]){
+                // Brightness Down
+                last_key = ODROID_INPUT_DOWN;
+                int8_t level = odroid_display_get_backlight();
+                if (level > 0) odroid_display_set_backlight(--level);
+            }
+            else if(joystick->values[ODROID_INPUT_A]){
+                // Save State
+                last_key = ODROID_INPUT_A;
+                odroid_system_emu_save_state(0);
+            }
+            else if(joystick->values[ODROID_INPUT_B]){
+                // Load State
+                last_key = ODROID_INPUT_B;
+                odroid_system_emu_load_state(0);
```

Comment:
It could be nice to get a dialog where you have to confirm the load so you don't accidentally load an old state. Or maybe that's gold plating, not sure.

Reply:
it's pretty hard to accidentally hit this button combo, but I somewhat agree with you; but I think also reloading from the normal menu should also require a confirmation screen (everyones definitely loaded a savestate by accident before and lost hours of progress). I'd say we can tackle that in a separate PR as this one is already pretty chunky.

Reply:
Yep totally agree. It was just something that popped up in my mind when reading this code.